src/state/logging-mixin.ts

```diff
@@ -21,6 +22,44 @@ export const loggingMixin = <T extends Constructor<HassBaseEl>>(
   superClass: T
 ) =>
   class extends superClass {
+    protected hassConnected() {
+      super.hassConnected();
```

Comment:
I think the reason we did this in core, was that we would catch error as soon and as low as possible, I see that that makes no sense because there is no connection yet, so they can not be send to core anyway, but I think the idea was that we would store those errors and send them as soon as there would be a connection, or if the error would be in the connection itself show it on screen.

Reply:
Yeah I thought about that as well, but I think the logging listeners should stay separate rather than complicating them with startup logic.  I think a follow up enhancement would be to add startup listeners early on, and then remove them here after the connection is complete.

Just storing them and expecting to log later is flawed since many such errors would likely prevent the connection from completing or even defining the custom element.  It's also not necessarily straightforward to handle.  For example, I tried to also test errors inside other mixins, but those didn't even get logged to the console let alone handled by the listeners.